docs/interface.rst

```diff
@@ -325,74 +325,50 @@ There is a WebSocket-based status API located at ``/v1/status``.
-
-The state for each folder consists of the following information:
-
-- ``"synchronizing"``: ``true`` or ``false``. When ``true`` the
-  magic-folder daemon is uploading data to or downloading data from
-  Tahoe-LAFS.
-- ``"tahoe"``: a dict containing status information about the Tahoe-LAFS connection
-  - ``"connected"``: the number of storage-servers the client is connected to
-  - ``"desired"``: the number of storage-servers we want to connect to
-  - ``"happy"``: ``true`` if ``"connected"`` is greater than the client's configured "happy"
-- ``"folders"`` contains keys mapping the folder name to:
-  - ``"uploads"`` and ``"downloads"`` contain currently queued or active uploads (or downloads). Each ``dict`` in these lists contain:
-
-    - ``"relpath"``: the relative-path
-    - ``"queued-at"``: the Unix timestamp when this item was queued
-    - ``"started-at"``: the Unix timestamp when we started uploading (or downloading) this item. This key will not exist until we do start.
-
-  - ``"recent"`` contains a list up to 30 of the most-recently updated files. Each ``dict`` in this list contains:
-
-    - ``"relpath"``: the relative path of this item
-    - ``"modified"``: the Unix timestamp when the on-disk file was most-recently modified
-    - ``"last-updated"``: the Unix timestamp when this item's state was updated in the magic-folder
-    - ``"conflicted"``: a boolean indicating if there is a conflict for this relative path
-  - ``"scanner"`` contains information about the local changes scanner
-    - ``"last-scan"``: ``null`` if no scan is completed yet, or the timestamp of the last completion
-  - ``"poller"`` contains information about the remote changes poller
-    - ``"last-poll"``: ``null`` if no scan is completed yet, or the timestamp of the last completion
-
-Clients should be tolerant of keys in the state they don't understand.
-Unknown state keys should be ignored.
-Note that ``"modified"`` is when the local state for thie item changed while ``"last-updated"`` is to do with the filesystem modification time.
-For example, a file may have an on-disk modification time that is older than the last time we updated our state about it, especially one that came from another device.
+...where the ``events`` list contains some non-zero number of event messages.
+The first message, upon connect, will likely contain many events: enough to give a consistent view of the current state.
+Thereafter, most messages will include only a single event (although clients should handle any number).
+
+Every event has a ``"kind"`` key describing what sort of message it is.
+Events will contain other keys; clients should be tolerant of keys in the state they don't understand.
 
 The client doesn't send any messages to the server; it is an error to do so.
+
+The follow event kinds are understood (see ``status.py`` for more details on the sending side, and ``cli.py`` for an example of receiving them):
```

Comment:
These seem reasonable (some of the present-tense stuff is a holdover from when it was "state" I guess and that made sense).

Shall I fix these up and merge...?

Reply:
>Shall I fix these up and merge...?

If you don't mind, please hold off for a little longer, as I'm still trying to unbreak my integration code/tests (which depended heavily on the old status message format) and narrow down what else might be missing or behaving differently...

One other issue I've noticed so far is that according to the docs, the new "error" event no longer provides the folder-name context. This prevents consumers of this new API from being able to tell which folder an error corresponds to. To elaborate, with the old "state"-based status API, "error" messages were scoped/nested beneath the name of folder that they corresponded to, like so:

```
{
  "state": {
    "synchronizing": False,
    "folders": {
      "Documents": {
        "uploads": [],
        "downloads": [],
        "errors": [
          {
            "timestamp": 1678901234,
            "summary": "Something went wrong!",
          }
        ],
        "recent": [],
        "tahoe": {"happy": True, "connected": 1, "desired": 1},
        "scanner": {"last-scan": 1678901234.5678},
        "poller": {"last-poll": 1678901234.5678},
        }
      }
   }
}
```

From this, it's easy/possible to tell that the "Something went wrong!" error occurred in the "Documents" folder and not some other folder. With the new "events"-based API though, we only get a `timestamp` and `summary` keys. It would be nice if the folder-name could be included in "error" event messages if/where the error is scoped to a particular folder (perhaps with a `None`/`null` value in the event that there is no corresponding folder?) or if there were two "error" message kinds -- for example a "folder-error-occurred" kind for errors that do correspond to a particular folder (and whose messages would always have a "folder" key/value) and a "general-error-occurred" kind for errors that do not.

Reply:
Yes, might be just a docs thing then. I think it's also _optional_ actually: there are errors that are possible that aren't scoped to a folder (e.g. "can't connect to tahoe" could be one).

Reply:
double-checking the code, the event API will only ever receive `error` events with the `folder` specified -- so it's just a doc issue then